pkg/render/apiserver.go

```diff
@@ -2023,3 +2128,51 @@ func (c *apiServerComponent) getDeprecatedResources() []client.Object {
 
 	return renamedRscList
 }
+
+func (cfg *APIServerConfiguration) IsSidecarInjectionEnabled() bool {
+	return cfg.ApplicationLayer != nil &&
+		cfg.ApplicationLayer.Spec.SidecarInjection != nil &&
+		*cfg.ApplicationLayer.Spec.SidecarInjection == operatorv1.SidecarEnabled
+}
+
+func (c *apiServerComponent) l7AdmissionControllerContainer() corev1.Container {
+	volumeMounts := []corev1.VolumeMount{
+		c.cfg.TLSKeyPair.VolumeMount(c.SupportedOSType()),
+	}
+
+	l7AdmssCtrl := corev1.Container{
+		Name:            L7AdmissionControllerContainerName,
```

Comment:
We need to add this name as a valid option in the API server overrides section: https://github.com/tigera/operator/blob/6eeefb15c96cfabe0ce747b053101726c1276df5/api/v1/apiserver_types.go#L77-L90

We should also add a UT that verifies it's possible to set resource requirements for this new container like these here: https://github.com/tigera/operator/blob/6eeefb15c96cfabe0ce747b053101726c1276df5/pkg/render/apiserver_test.go#L1025-L1034